pkg/api/volume.go

```diff
@@ -52,8 +52,10 @@ func (this *VolumePortal) CreateVolume() {
 		return
 	}
 
-	// Call global controller variable to handle create volume request.
-	result, err := controller.Brain.CreateVolume(c.GetContext(this.Ctx), &volume)
+	// NOTE:It will create a volume entry into the database and initialize its status
+	// as "creating". It will not wait for the real volume creation to complete
+	// and will return result immediately.
+	result, pool, profile, err := controller.Brain.CreateVolumeDBEntry(c.GetContext(this.Ctx), &volume)
```

Comment:
Like I said, the method between api package and controller package must be unique for every operation, you can't call two methods at the same time, and here are some reasons:
* Api package can't decide whether the operation is sync or not, it's all decided by controller package.
* Later if we decouple api and controller package, then we have to reduce the calling times because in your design it would take two calls if users want to create a volume.